packages/jaeger-ui/src/components/TracePage/index.tsx

```diff
@@ -395,15 +396,17 @@ export class TracePageImpl extends React.PureComponent<TProps, TState> {
             </section>
           ) : (
             <section style={{ paddingTop: headerHeight }}>
-              <TraceTimelineViewer
-                registerAccessors={this._scrollManager.setAccessors}
-                scrollToFirstVisibleSpan={this._scrollManager.scrollToFirstVisibleSpan}
-                findMatchesIDs={spanFindMatches}
-                trace={data}
-                updateNextViewRangeTime={this.updateNextViewRangeTime}
-                updateViewRangeTime={this.updateViewRangeTime}
-                viewRange={viewRange}
-              />
+              <UIElementsContext.Provider value={{ Popover }}>
```

Comment:
The `value` here should be a `const` at the top of the file. Something like `const UI_ELEMENTS = { Popover };`. Without that, ever render of `TracePage/index` will cause all consumers to re-render [(source)](https://reactjs.org/docs/context.html#caveats).